dasharo-security/measured-boot.robot

```diff
@@ -18,25 +18,376 @@ Resource            ../keys.robot
+    ...    actual PCRs values
+    ${tpm2_eventlog}=    Execute Command In Terminal
+    ...    tpm2_eventlog /sys/kernel/security/tpm0/binary_bios_measurements
+    Should Not Contain    ${tpm2_eventlog}    ERROR: Unable to run tpm2_eventlog
+    FOR    ${algo}    IN    sha1    sha256
+        ${eventlog_pcrs}=    Get PCRs From Eventlog    ${tpm2_eventlog}    ${algo}
+        FOR    ${pcr_element}    IN    @{eventlog_pcrs}
+            ${pcr}    ${hash}=    Split String    ${pcr_element}    separator=:
+            ${sha_hash}=    Execute Command In Terminal
+            ...    cat /sys/class/tpm/tpm0/pcr-${algo}/${pcr}
+            Should Contain    ${hash}    ${sha_hash}    ignore_case=${TRUE}
+        END
+    END
+
+MBO003.001 Changing Secure Boot certificate changes only PCR-7
+    [Documentation]    Check if changes to Secure Boot certificates influence
+    ...    PCR-7 value and only PCR-7
+    Skip If    not ${TESTS_IN_FIRMWARE_SUPPORT}    Tests in firmware are not supported
+    Skip If    not ${SECURE_BOOT_SUPPORT}    Secure Boot tests are not supported
+    Restore Secure Boot Defaults
+    ${sb_menu}=    Reenter Menu And Return Construction
+    Disable Secure Boot    ${sb_menu}
+    Save Changes And Reset
+
+    Boot System Or From Connected Disk    ubuntu
+    Login To Linux
+    Switch To Root User
+    ${default_hashes}=    Get PCRs State From Linux
+
+    Power On
+    ${sb_menu}=    Enter Secure Boot Menu And Return Construction
+    ${sb_menu}=    Enter Advanced Secure Boot Keys Management And Return Construction    ${sb_menu}
+    ${sb_menu}=    Enter Submenu From Snapshot And Return Construction    ${sb_menu}    DBX Options
+    ${sb_menu}=    Enter Submenu From Snapshot And Return Construction    ${sb_menu}    Delete Signature
+    Enter Submenu From Snapshot    ${sb_menu}    Delete All Signature List
+    Read From Terminal Until    Press 'Y' to delete signature list
+    Write Into Terminal    Y
+    Sleep    1s
+    Save Changes And Reset
+
+    Boot System Or From Connected Disk    ubuntu
+    Login To Linux
+    Switch To Root User
+    FOR    ${pcr_hash}    IN    @{default_hashes}
+        ${pcr}    ${hash}=    Split String    ${pcr_hash}    separator=:
+        ${new_hash}=    Execute Command In Terminal    cat ${pcr}
+        IF    '/7' in '${pcr}'
+            Should Not Be Equal    ${hash}    ${new_hash}
+        ELSE
+            Should Be Equal    ${hash}    ${new_hash}
+        END
+    END
+
+MBO004.001 Changing Dasharo network boot settings changes only PCR-1
+    [Documentation]    Check if changes to Dasharo security settings influence PCR-1
+    ...    value and only PCR-1
+    Skip If    not ${TESTS_IN_FIRMWARE_SUPPORT}    Tests in firmware are not supported
+    Skip If    not ${DASHARO_NETWORKING_MENU_SUPPORT}    Tests in Dasharo Networking Menu are not supported
+    Power On
+    Boot System Or From Connected Disk    ubuntu
+    Login To Linux
+    Switch To Root User
```

Comment:
Is there a reason to reset here and in 2 tests below?  If it's to guard against dbx changes, maybe test setup keyword should check current state of the service and if it's on disable it and reset SB state.  This way no test case should worry about dbx and there won't be unnecessary reboots.

Reply:
In this and similar tests I:
* login to linux and save current PCR values
* restart and change dasharo settings
* restart and login to linux to compare PCR values

I'm not sure If I could do this test with less restarts.

Reply:
My point is that at the beginning each of these tests Linux is already running and you have the shell, so why restart it before saving PCR values?

Reply:
Because if previous test fails somewhere in middle while it's in BIOS then next test would have no chance of passing.

Reply:
Then this is kind of a workaround rather than integral part of the test.  A comment that this guards against messed up state after the previous test would be helpful.

Reply:
Does it? I always thought that it was default that tests should be as self-contained as possible, especially since I don't think we have some default way to describe dependencies between test (check/skip if test x failed).

Reply:
Self-contained tests are great, but rebooting takes time and now tests are inconsistent: some start with a reset and some don't which made me wonder why the former need a reset.

Reply:
The only one that doesn't start with reset are the first 2 that only call commands in Linux. The rest are restarted either explicitly by Power On keyword or inside other keywords.